system/jlib/jtrace.cpp

```diff
@@ -730,6 +730,18 @@ IProperties * getClientHeaders(const ISpan * span)
     return headers.getClear();
 }
 
+void setSpanAttribute(ISpan * span, const char * name, const char * val)
```

Comment:
minor: Not sure why you would call this rather than span->setSpanAttribute()

Reply:
yeah I ended up going this route because in logging.cpp, ctx->queryContextLogger() returns a const, this approach seemed to bypass the const restriction on the ctxLogger. Do we want to make the contexlogger non-const?

Reply:
You can still call

        activeSpan->setSpanAttribute(name, value);

from within a const function.  const generally prevents direct member variables from being modified - it does not prevent objects pointed to by member variables from being modified.